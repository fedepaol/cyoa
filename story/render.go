package story

import (
	"html/template"
	"io"
)

type renderer struct {
	t *template.Template
}

type TemplateData struct {
	Node Node
	URL  template.URL
}

func newRenderer(templateFile string) *renderer {
	t := template.Must(template.ParseFiles(templateFile))
	return &renderer{t}
}

func (r *renderer) render(w io.Writer, baseURL string, n Node) error {
	err := r.t.Execute(w, TemplateData{n, template.URL(baseURL)})
	return err
}
