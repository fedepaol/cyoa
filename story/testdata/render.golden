<!DOCTYPE html>
<html>
  <head>
    <title>.Node.Title</title>
  </head>
  <body>
  <h1>Title</h1>
  
  Story
  Story1
   <a href=localhost:8080/bb>aa</a>
  </body>
</html>
